StickyTabBarViewController/Classes/MainTabBarController.swift

```diff
@@ -10,53 +10,52 @@ import UIKit
 public protocol StickyViewControllerSupporting: UITabBarController {
     var collapsedHeight: CGFloat { get set }
     var animationDuration: TimeInterval { get set }
-    func configureCollapsedViewController(withChildViewController childViewController: Expandable)
-    func removeCollapsibleViewController(animated: Bool)
-    func collapseCollapsibleViewController()
+    func configureCollapsableChild(_ childViewController: Expandable)
+    func removeCollapsableChild(animated: Bool)
+    func collapseChild()
 }
 
 open class StickyViewControllerSupportingTabBarController: UITabBarController, StickyViewControllerSupporting {
-    private var collapsableVCFlow: ExpandableViewController?
+    
+    // MARK: - Public properties
     public var collapsedHeight: CGFloat = 50.0
     public var animationDuration: TimeInterval = 0.5
     
-    final public func updateCollapsedHeight(to value: CGFloat) {
-        collapsedHeight = value
-    }
+    // MARK: - Private properties
+    private var collapsableVCFlow: ExpandableViewController?
     
-    final  public func updateAnimationDuration(to value: TimeInterval) {
-        animationDuration = value
-    }
```

Comment:
We actually need these methods to update values, I know they are ugly but I couldn't find another solution before. One can not simply subclass `StickyViewControllerSupportingTabBarController` and override variables of `collapsedHeight` and `animationDuration`. Attempting to do so, generates the following compiler error:  `Cannot override with a stored property 'collapsedHeight'` We could overcome this with an initializer but there are cases where Tab bar controller is not initialised through code. 

https://stackoverflow.com/questions/26691935/overriding-a-stored-property-in-swift

Reply:
I see! This actually begs the question on other APIs besides these ones. So I see a few things: 

- From the consumer perspective, sure, I'd expect these values (height & animation) to be configurable, but even after subclassing `StickyViewControllerSupportingTabBarController` you could still override their values in either a constructor (say `override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?)`) or another instance method (say viewDidLoad()) so it'd be perfectly allowed to set a new value there. 
- Does it make sense to only allow that configuration then? We're marking the remaining APIs as final so it's not possible to override them, so the configuration from the client side feels rather "limited" in that regard (although this does not necessarily mean it's needed)
 
> We could overcome this with an initializer but there are cases where Tab bar controller is not initialised through code.

This would be up to the consumer/client of the library, correct? 

I can bring them back for sure but I wanted to leave here some thoughts 👍

Reply:
I agree with all the points and I also didn't like having those methods honestly, I just got confused and didn't even try to set them somewhere else, I quickly tested in sample project so it is possible to override them as following:

```
required init?(coder: NSCoder) {
    super.init(coder: coder)
    collapsedHeight = 100.0
    animationDuration = 2.0
}
```

Or as you mentioned it is also possible on `viewDidLoad`:
```
override func viewDidLoad() {
    super.viewDidLoad()
    collapsedHeight = 100.0
    animationDuration = 2.0
}
```

For some reason it didnt work with init with nibname maybe because simple there is no nib file for tab bar controller in sample project. 

So it would be nice to update README with these changes and about how to use it, do you think it would make sense to update it in this PR or do it later on a separate PR? @Thurman1776-

Reply:
> For some reason it didnt work with init with nibname maybe because simple there is no nib file for tab bar controller in sample project.

I think this is the case. 

For the README, sure! I think I should update it here. Probably makes sense to have a rule to not merge PRs that added APIs changes without updating the README, otherwise it's very easy to overlook it & it will become outdated. 

What do you think? 
@emrepun

Reply:
Yes that rule would be very beneficial I agree. Should that rule be written as default on PR description screen when someone tries to create a PR? @Thurman1776- 

Also, for this PR I updated README and created a PR for this one: https://github.com/emrepun/StickyTabBarViewController/pull/10

Reply:
Probably as a default PR description would serve as a good reminder. Do you know how to do that? I've never done it but I assume it should be somewhere in the project settings.

Reply:
I will check it out @Thurman1776-  👍